cmd/peer-rest-client.go

```diff
@@ -423,51 +423,49 @@ func (client *peerRESTClient) SignalService(sig serviceSignal) error {
 }
 
 // Trace - send http trace request to peer nodes
-func (client *peerRESTClient) Trace(doneCh chan struct{}, trcAll bool) (chan []byte, error) {
-	ch := make(chan []byte)
+func (client *peerRESTClient) Trace(traceCh chan interface{}, doneCh chan struct{}, trcAll bool) {
 	go func() {
-		cleanupFn := func(cancel context.CancelFunc, ch chan []byte, respBody io.ReadCloser) {
-			close(ch)
-			if cancel != nil {
-				cancel()
-			}
-			http.DrainBody(respBody)
-		}
 		for {
 			values := make(url.Values)
 			values.Set(peerRESTTraceAll, strconv.FormatBool(trcAll))
+
 			// get cancellation context to properly unsubscribe peers
 			ctx, cancel := context.WithCancel(context.Background())
 			respBody, err := client.callWithContext(ctx, peerRESTMethodTrace, values, nil, -1)
```

Comment:
Additionally we need to close the response body as well before each retry iteration.